src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt

```diff
@@ -39,33 +41,47 @@ class AttemptSnapshotStep(
     val clusterService: ClusterService,
     val client: Client,
     val config: SnapshotActionConfig,
-    managedIndexMetaData: ManagedIndexMetaData
+    managedIndexMetaData: ManagedIndexMetaData,
+    settings: Map<String, Any>
```

Comment:
clusterService.clusterSettings.get(SNAPSHOT_DENY_LIST)

Reply:
Let's just directly access settings using above example instead of passing down a settings map to all the steps

Reply:
change to directly access